sql/size_utils.sql

```diff
@@ -394,6 +394,86 @@ $BODY$ SET search_path TO pg_catalog, pg_temp;
 -- Estimated number of rows according to catalog tables
 CREATE OR REPLACE FUNCTION @extschema@.approximate_row_count(relation REGCLASS)
 RETURNS BIGINT
+LANGUAGE PLPGSQL VOLATILE STRICT AS
+$BODY$
+DECLARE
+    local_table_name       NAME = NULL;
+    local_schema_name      NAME = NULL;
+    is_distributed   BOOL = FALSE;
+    is_compressed    BOOL = FALSE;
+    uncompressed_row_count BIGINT = 0;
+    compressed_row_count BIGINT = 0;
+    local_compressed_hypertable_id INTEGER = 0;
+    local_compressed_chunk_id INTEGER = 0;
+    compressed_hypertable_oid  OID;
+    local_compressed_chunk_oid  OID;
+    max_compressed_row_count BIGINT = 1000;
+    is_compressed_chunk INTEGER;
+BEGIN
+    SELECT relname, nspname FROM pg_class c
+    INNER JOIN pg_namespace n ON (n.OID = c.relnamespace)
+    INTO local_table_name, local_schema_name
+    WHERE c.OID = relation;
+
+    -- If OID is Hypertable
+    IF EXISTS (SELECT 1
+               FROM _timescaledb_catalog.hypertable WHERE table_name = local_table_name AND schema_name = local_schema_name) THEN
+        SELECT compressed_hypertable_id FROM _timescaledb_catalog.hypertable INTO local_compressed_hypertable_id
+        WHERE table_name = local_table_name AND schema_name = local_schema_name;
+        IF local_compressed_hypertable_id IS NOT NULL THEN
+           uncompressed_row_count = _timescaledb_internal.get_approx_row_count(relation);
+
+           WITH compressed_hypertable AS (SELECT table_name, schema_name FROM _timescaledb_catalog.hypertable ht
+           WHERE ht.id = local_compressed_hypertable_id)
+           SELECT c.oid INTO compressed_hypertable_oid FROM pg_class c
+           INNER JOIN compressed_hypertable h ON (c.relname = h.table_name)
+           INNER JOIN pg_namespace n ON (n.nspname = h.schema_name);
+
+           compressed_row_count = _timescaledb_internal.get_approx_row_count(compressed_hypertable_oid);
+           RETURN (uncompressed_row_count + (compressed_row_count * max_compressed_row_count));
+        ELSE
+           uncompressed_row_count = _timescaledb_internal.get_approx_row_count(relation);
+           RETURN uncompressed_row_count;
+        END IF;
+    END IF;
+    -- IF OID is CHUNK
+    IF EXISTS (SELECT 1 FROM _timescaledb_catalog.chunk WHERE table_name = local_table_name AND schema_name = local_schema_name) THEN
+        with compressed_chunk as (select 1 as is_compressed_chunk from _timescaledb_catalog.chunk c
+        inner join _timescaledb_catalog.hypertable h on (c.hypertable_id = h.compressed_hypertable_id)
+        where c.table_name = local_table_name and c.schema_name = local_schema_name ),
+        chunk_temp as (select compressed_chunk_id from _timescaledb_catalog.chunk c where c.table_name = local_table_name and c.schema_name = local_schema_name)
```

Comment:
The chunk name + schema are unique in the database, right? Maybe you could remove the join with `hypertable` and combine this in a single select then.

Reply:
Actually, this query is meant to initialize necessary local variables required to process further cases. Depending upon the cases respective local variables are utilized. Join with hypertable is required to confirm that the chunk hypertable_id belongs to compressed_hypertable_id.